gasprice/feehistory_test.go

```diff
@@ -0,0 +1,112 @@
+package gasprice
+
+import (
+	"errors"
+	"math/big"
+	"testing"
+
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGasHelper_FeeHistory(t *testing.T) {
+	t.Parallel()
+
+	var cases = []struct {
+		Name                  string
+		ExpectedOldestBlock   uint64
+		ExpectedBaseFeePerGas []uint64
+		ExpectedGasUsedRatio  []float64
+		ExpectedRewards       [][]uint64
+		BlockRange            uint64
+		NewestBlock           uint64
+		RewardPercentiles     []float64
+		Error                 bool
+		GetBackend            func() Blockchain
+	}{
+		{
+			Name:              "Block does not exist",
+			Error:             true,
+			BlockRange:        10,
+			NewestBlock:       30,
+			RewardPercentiles: []float64{15, 20},
+			GetBackend: func() Blockchain {
+				header := &types.Header{
+					Number: 0,
+					Hash:   types.StringToHash("some header"),
+				}
+				backend := new(backendMock)
+				backend.On("Header").Return(header)
+				backend.On("GetBlockByNumber", mock.Anything, true).Return(errors.New("block not found"))
```

Comment:
I think you need to do this:

```suggestion
				backend.On("GetBlockByNumber", mock.Anything, mock.Anything).Return(nil, false)
```